hcn/hcnnamespace.go

```diff
@@ -307,6 +309,19 @@ func GetNamespaceContainerIds(namespaceID string) ([]string, error) {
 	return containerIds, nil
 }
 
+func CanRemovePauseContainer() bool {
```

Comment:
I don't think this function makes sense to put in `hcn`, since it's doing something very specific to our container implementation. I would recommend putting it in pod.go.

Reply:
Actually I had missed we were calling this below. Can we structure this in such a way that the shim code is what determines how to set up the pause container networking, rather than relying on `hcn` to do it automatically?

Reply:
CanRemovePauseContainer() is checking for the HNS version and that's the reason I had it in the hcn package. Its used to ensure that HNS version is one that supports removal of pause containers. Decision to remove pause container is happening in pod.go . Am I misunderstanding something you are suggesting?

Reply:
My overall point is I don't think the `hcn` package should know what a "pause container" is. It should deal only with HCN-level constructs.

Reply:
e.g. should containerd's `pkg/netns` be the one setting `ReadOnCreate`? Or should package `hcn` do it internally, but without exposing a function referring to pause containers?